### PR TITLE
RUM-3338 Allow disabling ANR tracking

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Add a configuration option to disable ANR tracking in the RUM feature configuration.

### Motivation

Our in-house ANR tracking provides information that are an estimation and not exactly the same as what the Google Play Store console reports. This makes it an optional feature to let customer disable it.

### CODE REVIEW

**Comment by 0xnm** [comment]
This goes in parallel with the work about ANR changes (different way to detect them for Android 11 and above, introduction of fatal vs non-fatal ANRs which will bring new configuration method), so I would propose to wait a bit so that all these changes are aligned.

File: features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt [hunk 1]
```diff
@@ -242,6 +242,17 @@ data class RumConfiguration internal constructor(
             return this
         }
 
+        /**
+         * Enable or disable ANR tracking.
+         * This tracking uses a custom heuristic to estimate ANRs, but might give different result than the
+         * Google Play Store.
+         * @param enabled whether ANR tracking should be enabled (default: true)
+         */
+        fun trackApplicationNotResponding(enabled: Boolean): Builder {
```
Comment (mariusc83) [inline features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/RumConfiguration.kt:251]: don't you need to add this also in the `TelemetryConfiguration` event ?

**Comment by 0xnm** [comment]
@xgouchet probably this one can be closed in favour of https://github.com/DataDog/dd-sdk-android/pull/1914

**Comment by xgouchet** [comment]
Yep Agreed

**Pull request closed without merging**